tests/components/withings/__init__.py

```diff
@@ -1 +1,85 @@
 """Tests for the withings component."""
+from collections.abc import Iterable
+from typing import Any, Optional
+from urllib.parse import urlparse
+
+import arrow
+from withings_api import DateType
+from withings_api.common import (
+    GetSleepSummaryField,
+    MeasureGetMeasGroupCategory,
+    MeasureGetMeasResponse,
+    MeasureType,
+    SleepGetSummaryResponse,
+    UserGetDeviceResponse,
+)
+
+from homeassistant.components.webhook import async_generate_url
+from homeassistant.core import HomeAssistant
+
+from .common import ProfileConfig, WebhookResponse
+
+
+async def call_webhook(
+    hass: HomeAssistant, webhook_id: str, data: dict[str, Any], client
+) -> WebhookResponse:
+    """Call the webhook."""
+    webhook_url = async_generate_url(hass, webhook_id)
+
+    resp = await client.post(
+        urlparse(webhook_url).path,
+        data=data,
+    )
+
+    # Wait for remaining tasks to complete.
+    await hass.async_block_till_done()
+
+    data: dict[str, Any] = await resp.json()
+    resp.close()
+
+    return WebhookResponse(message=data["message"], message_code=data["code"])
+
+
+class MockWithings:
```

Comment:
Could we not mock the library instead of creating a complete mocking class?

Reply:
I already wrote a new library for withings, but a lot is Happening in this integration that I just created this to be quick with testing